pkg/workflows/workflow.go

```diff
@@ -566,3 +566,35 @@ func (w *workflowLifecycle) injectActiveDeadlineSeconds(wf *unstructured.Unstruc
 
 	return nil
 }
+
+func IsValidV1WorkFlow(obj interface{}) error {
+	wf, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("error : unstructured type.")
+	}
+
+	if wf.GetKind() != "Workflow" || wf.GetAPIVersion() != "argoproj.io/v1alpha1" {
```

Comment:
Could we do prefix matching of "argoproj.io/v1" if our intention is to support workflow v1?
argo workflow could add "argoproj.io/v1alpha2" or ""argoproj.io/v1beta1" in the future.

cc @kevdowney

Reply:
Actually I should take my comment back, specify the api version explicitly is better.

Reply:
Each release, match one argo controller version, it is constant.